src/@next/Banner/Banner.tsx

```diff
@@ -58,37 +61,60 @@ export const Banner = React.forwardRef<HTMLDivElement, BannerProps>(
         ref={ref}
         data-titled={!!title}
         data-status={status}
+        data-type={type}
         {...props}
       >
-        <StyledBannerTitleContainer>
-          <Icon name={iconNameValue} />
-          {title && (
-            <StyledBannerTitle>
-              <Typography as="span" variant="subtitle1">
-                {title}
-              </Typography>
-            </StyledBannerTitle>
-          )}
-          {dismissable && (
-            <StyledCloseIconWrapper role="button" onClick={() => onDismiss?.()}>
-              <Icon name="ri-close" />
-            </StyledCloseIconWrapper>
-          )}
-        </StyledBannerTitleContainer>
-        <StyledBannerContentContainer>
+        {type === 'static' ? (
+          <StyledBannerTitleContainer>
+            <Icon name={iconNameValue} />
+            {title && (
+              <StyledBannerTitle data-type={type}>
+                <Typography as="span" variant="subtitle1">
+                  {title}
+                </Typography>
+              </StyledBannerTitle>
+            )}
+            {dismissable && (
+              <StyledCloseIconWrapper
+                role="button"
+                onClick={() => onDismiss?.()}
+              >
+                <Icon name="ri-close" />
+              </StyledCloseIconWrapper>
+            )}
+          </StyledBannerTitleContainer>
+        ) : (
+          type === 'fixed' && (
+            <StyledBannerContentContainer data-type={type}>
+              <Icon name={iconNameValue} />
+            </StyledBannerContentContainer>
+          )
+        )}
+        <StyledBannerContentContainer data-type={type}>
           <Typography as="div" variant="body1">
             {children}
           </Typography>
         </StyledBannerContentContainer>
-        <StyledBannerContentContainer>
-          {action ||
-            (secondaryAction && (
+        <StyledBannerContentContainer data-type={type}>
+          {action &&
+            (secondaryAction !== 'undefined' ? (
               <ButtonGroup>
                 {action}
                 {secondaryAction}
               </ButtonGroup>
+            ) : (
+              { action }
             ))}
         </StyledBannerContentContainer>
```

Comment:
preety sure this was buggy before (?)

Before, if you have both `action` and `secondaryAction`, you can only see the `action` and not the `secondaryAction`. So essentially anytime action is defined, then render action; otherwise if it is not defined and secondary action is defined then render both, which I think is a wrong logic. Fixed this issue.

Reply:
Hmmm wouldn't  undefined is valid value in jsx? It just won't render anything right? Is there a test cases for rendering action and secondAction?

Reply:
I saw the visual tests, preety sure no tests involving 2 actions are there yet;

Yes, undefined should be valid. But the logic was something like this:
```
<StyledBannerContentContainer>
    {action ||
      (secondaryAction && (
        <ButtonGroup>
          {action}
          {secondaryAction}
        </ButtonGroup>
      ))}
</StyledBannerContentContainer>
```

cmiiw, I think with that logic:
- If action is not undefined, render action
- If action is undefined and if secondaryAction is defined, render button group

The problem here is that if action and secondaryAction is both defined, it will just render action and not the button group with both actions.

Reply:
Perhaps after this, I'll also be creating the visual test cases for the banner with two actions too

Reply:
Oh you're right, because the the secondary action statement is not part of the condition, good catch, I think we should hoist the conditions like

```
const hasActions = action || secondaryAction;

<StyledBannerContentContainer>
    {hasActions && (
        <ButtonGroup>
          {action}
          {secondaryAction}
        </ButtonGroup>
      )}
</StyledBannerContentContainer>
```